subscription.go

```diff
@@ -149,10 +150,27 @@ func (s *Subscription) Unmonitor(monitoredItemIDs ...uint32) (*ua.DeleteMonitore
 		MonitoredItemIDs: monitoredItemIDs,
 		SubscriptionID:   s.SubscriptionID,
 	}
+
 	var res *ua.DeleteMonitoredItemsResponse
 	err := s.c.Send(req, func(v interface{}) error {
 		return safeAssign(v, &res)
 	})
+
+	if err == nil {
+		// remove monitored items
+		// todo(fs): should we guard this with a lock?
+		var items []*monitoredItem
+		for _, id := range monitoredItemIDs {
+			for _, item := range s.items {
```

Comment:
the nested loop here could make this expensive. as long as order doesn't matter, we could change `s.items` to `map[uint32]*monitoredItem`

Reply:
```
Running tool: /Users/frank/sdk/go1.17.1/bin/go test -benchmem -run=^$ -bench ^BenchmarkUnmonitorItems$ github.com/gopcua/opcua

goos: darwin
goarch: arm64
pkg: github.com/gopcua/opcua
BenchmarkUnmonitorItems/slice-8         	54013098	        25.24 ns/op	      23 B/op	       0 allocs/op
--- BENCH: BenchmarkUnmonitorItems/slice-8
    subscription_test.go:29: src 1 dst 0
    subscription_test.go:29: src 100 dst 50
    subscription_test.go:29: src 10000 dst 5000
    subscription_test.go:29: src 1000000 dst 500000
    subscription_test.go:29: src 54013098 dst 27006549
BenchmarkUnmonitorItems/map-8           	32447011	        35.72 ns/op	       0 B/op	       0 allocs/op
--- BENCH: BenchmarkUnmonitorItems/map-8
    subscription_test.go:53: src 0
    subscription_test.go:53: src 50
    subscription_test.go:53: src 5000
    subscription_test.go:53: src 500000
    subscription_test.go:53: src 16223505
PASS
ok  	github.com/gopcua/opcua	15.666s
```

Reply:
With pre-alloc 

```
Running tool: /Users/frank/sdk/go1.17.1/bin/go test -benchmem -run=^$ -bench ^BenchmarkUnmonitorItems$ github.com/gopcua/opcua

goos: darwin
goarch: arm64
pkg: github.com/gopcua/opcua
BenchmarkUnmonitorItems/slice-8         	51153620	        24.03 ns/op	      20 B/op	       0 allocs/op
--- BENCH: BenchmarkUnmonitorItems/slice-8
    subscription_test.go:29: src 1 dst 0
    subscription_test.go:29: src 100 dst 50
    subscription_test.go:29: src 10000 dst 5000
    subscription_test.go:29: src 1000000 dst 500000
    subscription_test.go:29: src 51153620 dst 25576810
BenchmarkUnmonitorItems/slice_pre-alloc-8         	91635986	        22.77 ns/op	       8 B/op	       0 allocs/op
--- BENCH: BenchmarkUnmonitorItems/slice_pre-alloc-8
    subscription_test.go:51: src 1 dst 0
    subscription_test.go:51: src 100 dst 50
    subscription_test.go:51: src 10000 dst 5000
    subscription_test.go:51: src 1000000 dst 500000
    subscription_test.go:51: src 91635986 dst 45817993
BenchmarkUnmonitorItems/map-8                     	39885550	        43.72 ns/op	       0 B/op	       0 allocs/op
--- BENCH: BenchmarkUnmonitorItems/map-8
    subscription_test.go:75: src 0
    subscription_test.go:75: src 50
    subscription_test.go:75: src 5000
    subscription_test.go:75: src 500000
    subscription_test.go:75: src 19942775
PASS
ok  	github.com/gopcua/opcua	116.192s
```